src/storage/create_table/iceberg_create_table_request.cpp

```diff
@@ -0,0 +1,175 @@
+		} else {
+			auto list_type_obj = yyjson_mut_obj_add_obj(doc, field_obj, "element");
+			AddUnnamedField(doc, list_type_obj, *list_type);
+		}
+		yyjson_mut_obj_add_bool(doc, field_obj, "element-required", false);
+		return;
+	}
+	case LogicalTypeId::MAP: {
+		yyjson_mut_obj_add_strcpy(doc, field_obj, "type", "map");
+		D_ASSERT(column.children.size() == 2);
+		auto &key_child = column.children[0];
+		if (key_child->IsPrimitiveType()) {
+			yyjson_mut_obj_add_strcpy(doc, field_obj, "key",
+			                          IcebergTypeHelper::LogicalTypeToIcebergType(key_child->type).c_str());
+		} else {
+			auto key_obj = yyjson_mut_obj_add_obj(doc, field_obj, "key");
+			yyjson_mut_obj_add_strcpy(doc, key_obj, "type",
+			                          IcebergTypeHelper::LogicalTypeToIcebergType(key_child->type).c_str());
+			auto nested_key_fields_arr = yyjson_mut_obj_add_arr(doc, key_obj, "fields");
+			AddNamedField(doc, nested_key_fields_arr, *key_child);
+		}
+		yyjson_mut_obj_add_uint(doc, field_obj, "key-id", key_child->id);
+		auto &val_child = column.children[1];
+		if (val_child->IsPrimitiveType()) {
+			yyjson_mut_obj_add_strcpy(doc, field_obj, "value",
+			                          IcebergTypeHelper::LogicalTypeToIcebergType(val_child->type).c_str());
+		} else {
+			auto val_obj = yyjson_mut_obj_add_obj(doc, field_obj, "value");
+			yyjson_mut_obj_add_strcpy(doc, val_obj, "type",
+			                          IcebergTypeHelper::LogicalTypeToIcebergType(val_child->type).c_str());
+			auto nested_key_fields_arr = yyjson_mut_obj_add_arr(doc, val_obj, "fields");
+			AddNamedField(doc, nested_key_fields_arr, *val_child);
+		}
+		yyjson_mut_obj_add_uint(doc, field_obj, "value-id", val_child->id);
+		yyjson_mut_obj_add_bool(doc, field_obj, "value-required", false);
+		break;
+	}
+	default:
+		throw NotImplementedException("Unrecognized nested type %s", LogicalTypeIdToString(column.type.id()));
+	}
+	// skip doc, initial_default, and write_default for now.
+	//	yyjson_mut_obj_add_strcpy(doc, field_obj, "doc", "string");
+	//	yyjson_mut_obj_add_bool(doc, field_obj, "initial_default", true);
+	//	yyjson_mut_obj_add_bool(doc, field_obj, "write_default", true);
+}
+
+shared_ptr<IcebergTableSchema> IcebergCreateTableRequest::CreateIcebergSchema(const ICTableEntry *table_entry) {
+	auto schema = make_shared_ptr<IcebergTableSchema>();
+	// should this be a different schema id?
+	schema->schema_id = table_entry->table_info.table_metadata.current_schema_id;
+
+	// TODO: this can all be refactored out
+	//  this makes the IcebergTableSchema, and we use that to dump data to JSON.
+	//  we can just directly dump it to json.
+	auto column_iterator = table_entry->GetColumns().Logical();
+	idx_t field_id = 1;
+
+	auto next_field_id = [&field_id]() -> idx_t {
+		return field_id++;
+	};
+
+	for (auto column = column_iterator.begin(); column != column_iterator.end(); ++column) {
+		auto name = (*column).Name();
+		// TODO: is this correct?
```

Comment:
I think so, the ColumnDefinition doesn't have constraints, and `required=true` would only be set if there was a `NOT NULL` constraint on the column.